### PR TITLE
src/p11_ec.c: make additional check for EC_POINT

### PR DESCRIPTION
**Pull request by aalba6675**
[UPDATE] the AppVeyor failure doesn't seem to be related to the source code change

-------------

TL;DR Add a non-standard check for EC_POINT in a EC private key to support vendor-optimization where private key has EC_PARAMS and EC_POINT attributes.

1. For EC private keys we make a round-trip to the HSM to get EC_PARAMS and EC_POINT by looking for a corresponding public key object.
1. Some non-standard but helpful modules include EC_PARAMS/EC_POINT with the private key (SafeNet)
1. We already have a check for EC_PARAMS in the code (irregardless of public or private key added by  @mtrojnar), so I hope it is not too intrusive to make one more check for EC_POINT to support this nice vendor-optimization (saving a round-trip to the HSM)
1. This optimization is useful as it saves having an extra (relatively useless, duplicates information in a cert, for example) object on token ($$$)

Note: [Vendor Quirk] This is what AWS CloudHSM engine is doing: without a corresponding EC public key object OpenSSL apps like `req` and `pkey -pubout` work as the engine has filled in the public key attribute using the quirk of the private key

With engine_pkcs11, both these operations segfault (as expected) as the public key attribute is NULL.

The investigation is at #209.

### CODE REVIEW

**Comment by mtrojnar** [comment]
Can you please check if https://github.com/OpenSC/libp11/commit/281ccb3d60e77366bd6ece4b8a18d5ef28cb4f45 does the job for you?

**Comment by aalba6675** [comment]
I can confirm that this commit works.

**Comment by mtrojnar** [comment]
Thank you.
